ethcore/src/ethereum/ethash.rs

```diff
@@ -119,10 +128,27 @@ impl Engine for Ethash {
 							(header.gas_used * 6.into() / 5.into()) / bound_divisor))
 			}
 		};
+		if header.number >= self.ethash_params.dao_hardfork_transition &&
+			header.number <= self.ethash_params.dao_hardfork_transition + 9 {
```

Comment:
Might cause an overflow panic if header number is `0xffffffffffffffffff`. Is it guaranteed that no invalid headers are input to this function?